### PR TITLE
Fixes optional behaviour in navbar

### PR DESCRIPTION
**Pull request by guillaumegarcia13**
Simply apply 'optional' class to optional steps.

### CODE REVIEW

File: src/components/wizard-navigation-bar.component.html [hunk 1]
```diff
@@ -10,7 +10,8 @@
         done: isDone(step),
         editing: isEditing(step),
         optional: isOptional(step),
-        navigable: isNavigable(step)
+        navigable: isNavigable(step),
+        completion: isCompletionStep(step)
```
Comment (madoar) [inline src/components/wizard-navigation-bar.component.html]: You've added a new css class `completion` to the `li` html elements.
What is missing in my opinion are stylesheets making use of the new css class in https://github.com/madoar/angular-archwizard/blob/master/src/components/wizard-navigation-bar.component.horizontal.less and https://github.com/madoar/angular-archwizard/blob/master/src/components/wizard-navigation-bar.component.vertical.less

File: src/components/wizard-navigation-bar.component.ts [hunk 2]
```diff
@@ -69,6 +70,16 @@ export class WizardNavigationBarComponent {
...
   }
 
+  /**
+   * Checks, whether a [[WizardStep]] can be marked as `completion` in the navigation bar
+   *
+   * @param wizardStep The wizard step to be checked
+   * @returns True if the step is a completion step
+   */
+  public isCompletionStep(wizardStep: WizardStep) {
+    return wizardStep instanceof WizardCompletionStepComponent;
```
Comment (madoar) [inline src/components/wizard-navigation-bar.component.ts]: I don't understand, why we need to add a css tag to all completion steps of the wizard.
I mean normally we want to detect if a wizard has been completed, not if a wizard step is a completion step, right?

The first can be checked by using `wizardState.completed`.

File: src/components/wizard-navigation-bar.component.ts [hunk 3]
```diff
@@ -116,7 +127,7 @@ export class WizardNavigationBarComponent {
    * @returns True if the step can be marked as optional
    */
   public isOptional(wizardStep: WizardStep): boolean {
-    return wizardStep.optional && !wizardStep.completed && !wizardStep.selected && !this.wizardState.completed
+    return wizardStep.optional
```
Comment (madoar) [inline src/components/wizard-navigation-bar.component.ts]: Please add a `;` to the end of this line.

File: src/components/wizard-navigation-bar.component.html (hunk 1 above)
Comment (guillaumegarcia13) [reply src/components/wizard-navigation-bar.component.html]: It's a leftover from trials. It's useless, I propose to remove it, especially if it causes issues.

File: src/components/wizard-navigation-bar.component.ts (hunk 2 above)
Comment (guillaumegarcia13) [reply src/components/wizard-navigation-bar.component.ts]: It's a leftover from trials. It's useless, I propose to remove it, especially if it causes issues.

File: src/components/wizard-navigation-bar.component.ts (hunk 3 above)
Comment (guillaumegarcia13) [reply src/components/wizard-navigation-bar.component.ts]: I will.
In my defense, there was none before

**Pull request closed without merging**